From 9f53c93ef967537e9e6cd89b098a223088de3d33 Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jpirko@redhat.com>
Date: Wed, 13 Aug 2014 11:45:41 -0400
Subject: [net] iptunnels: remove net arg from iptunnel_xmit()

Message-id: <1407930342-13540-1-git-send-email-jpirko@redhat.com>
Patchwork-id: 87678
O-Subject: [patch rhel7 net 1/2] iptunnels: remove net arg from iptunnel_xmit()
Bugzilla: 1081956
RH-Acked-by: Jiri Benc <jbenc@redhat.com>
RH-Acked-by: Hannes Frederic Sowa <hannes@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>

BZ1081956
https://brewweb.devel.redhat.com/taskinfo?taskID=7837924
Tested on my testing kvm guest

upstream commit 8b7ed2d91d6afb0b55ba75f94b66e51f70783a46
Author: Nicolas Dichtel <nicolas.dichtel@6wind.com>
Date:   Mon Sep 2 15:34:54 2013 +0200

    iptunnels: remove net arg from iptunnel_xmit()

    This argument is not used, let's remove it.

Signed-off-by: Jiri Pirko <jpirko@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/vxlan.c b/drivers/net/vxlan.c
index b3bf445..3a9ac18 100644
--- a/drivers/net/vxlan.c
+++ b/drivers/net/vxlan.c
@@ -1645,8 +1645,7 @@ int vxlan_xmit_skb(struct net *net, struct vxlan_sock *vs,
  if (err)
   return err;
 
- return iptunnel_xmit(net, rt, skb, src, dst,
-   IPPROTO_UDP, tos, ttl, df);
+ return iptunnel_xmit(rt, skb, src, dst, IPPROTO_UDP, tos, ttl, df);
 }
 EXPORT_SYMBOL_GPL(vxlan_xmit_skb);
 
diff --git a/include/net/ip_tunnels.h b/include/net/ip_tunnels.h
index 9109f12..42968d1 100644
--- a/include/net/ip_tunnels.h
+++ b/include/net/ip_tunnels.h
@@ -156,8 +156,7 @@ static inline u8 ip_tunnel_ecn_encap(u8 tos, const struct iphdr *iph,
 }
 
 int iptunnel_pull_header(struct sk_buff *skb, int hdr_len, __be16 inner_proto);
-int iptunnel_xmit(struct net *net, struct rtable *rt,
-    struct sk_buff *skb,
+int iptunnel_xmit(struct rtable *rt, struct sk_buff *skb,
     __be32 src, __be32 dst, __u8 proto,
     __u8 tos, __u8 ttl, __be16 df);
 
diff --git a/net/ipv4/ip_tunnel.c b/net/ipv4/ip_tunnel.c
index 275c167..89d83f3 100644
--- a/net/ipv4/ip_tunnel.c
+++ b/net/ipv4/ip_tunnel.c
@@ -659,8 +659,7 @@ void ip_tunnel_xmit(struct sk_buff *skb, struct net_device *dev,
   }
  }
 
- err = iptunnel_xmit(tunnel->net, rt, skb,
-       fl4.saddr, fl4.daddr, protocol,
+ err = iptunnel_xmit(rt, skb, fl4.saddr, fl4.daddr, protocol,
        ip_tunnel_ecn_encap(tos, inner_iph, skb), ttl, df);
  iptunnel_xmit_stats(err, &dev->stats, dev->tstats);
 
diff --git a/net/ipv4/ip_tunnel_core.c b/net/ipv4/ip_tunnel_core.c
index 88e0d34..18f76d1 100644
--- a/net/ipv4/ip_tunnel_core.c
+++ b/net/ipv4/ip_tunnel_core.c
@@ -46,8 +46,7 @@
 #include <net/netns/generic.h>
 #include <net/rtnetlink.h>
 
-int iptunnel_xmit(struct net *net, struct rtable *rt,
-    struct sk_buff *skb,
+int iptunnel_xmit(struct rtable *rt, struct sk_buff *skb,
     __be32 src, __be32 dst, __u8 proto,
     __u8 tos, __u8 ttl, __be16 df)
 {
diff --git a/net/ipv6/sit.c b/net/ipv6/sit.c
index 05f84fd..3acf559 100644
--- a/net/ipv6/sit.c
+++ b/net/ipv6/sit.c
@@ -892,8 +892,8 @@ static netdev_tx_t ipip6_tunnel_xmit(struct sk_buff *skb,
   goto out;
  }
 
- err = iptunnel_xmit(dev_net(dev), rt, skb, fl4.saddr, fl4.daddr,
-       IPPROTO_IPV6, tos, ttl, df);
+ err = iptunnel_xmit(rt, skb, fl4.saddr, fl4.daddr, IPPROTO_IPV6, tos,
+       ttl, df);
  iptunnel_xmit_stats(err, &dev->stats, dev->tstats);
  return NETDEV_TX_OK;
 
diff --git a/net/openvswitch/vport-gre.c b/net/openvswitch/vport-gre.c
index 98c1428..3d147bd 100644
--- a/net/openvswitch/vport-gre.c
+++ b/net/openvswitch/vport-gre.c
@@ -175,7 +175,7 @@ static int gre_tnl_send(struct vport *vport, struct sk_buff *skb)
 
  skb->local_df = 1;
 
- return iptunnel_xmit(net, rt, skb, fl.saddr,
+ return iptunnel_xmit(rt, skb, fl.saddr,
         OVS_CB(skb)->tun_key->ipv4_dst, IPPROTO_GRE,
         OVS_CB(skb)->tun_key->ipv4_tos,
         OVS_CB(skb)->tun_key->ipv4_ttl, df);
-- 
1.7.1